applications/asset_tracker_v2/tests/lwm2m_codec/src/lwm2m_codec_test.c

```diff
@@ -0,0 +1,824 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <unity.h>
+#include <stdbool.h>
+#include <string.h>
+#include <errno.h>
+#include <net/lwm2m.h>
+#include <net/lwm2m_client_utils.h>
+
+#include "lwm2m_client_utils/mock_lwm2m_client_utils.h"
+#include "lwm2m/mock_lwm2m.h"
+#include "lwm2m/mock_lwm2m_resource_ids.h"
+#include "lte_lc/mock_lte_lc.h"
+#include "date_time/mock_date_time.h"
+
+#include "cloud_codec.h"
+#include "lwm2m_codec_defines.h"
+
+/* Expose internal variables in the uut for testing purposes. */
+extern uint8_t bearers[2];
+extern int battery_voltage;
+extern int32_t pressure_ts;
+extern int32_t temperature_ts;
+extern int32_t humidity_ts;
+extern int32_t button_ts;
+extern int64_t current_time;
+
+/* Minimum and maximum values for the BME680 present on the Thingy:91. */
+extern double temp_min_range_val;
+extern double temp_max_range_val;
+extern double humid_min_range_val;
+extern double humid_max_range_val;
+extern double pressure_min_range_val;
+extern double pressure_max_range_val;
+
+extern int config_update_cb(uint16_t obj_inst_id, uint16_t res_id, uint16_t res_inst_id,
+			    uint8_t *data, uint16_t data_len, bool last_block, size_t total_size);
+
+extern int object_path_list_add(struct cloud_codec_data *output, const char * const path[],
+				size_t path_size);
+
+extern int unity_main(void);
+extern int generic_suiteTearDown(int num_failures);
+
+/* Handler used to verify correct config update in the
+ * test_lwm2m_codec_config_update test case.
+ */
+static void cloud_codec_event_handler(const struct cloud_codec_evt *evt)
+{
+	TEST_ASSERT_EQUAL(CLOUD_CODEC_EVT_CONFIG_UPDATE, evt->type);
+	TEST_ASSERT_EQUAL(30, evt->config_update.gnss_timeout);
```

Comment:
I sugggest you create macros for these test values.
For example,

#define TEST_GNSS_TIMEOUT_VAL 30
#define TEST_ACTIVE_WAIT_TIMEOUT 120

That way you can use the macro both here in the stub and in the test case.